pkg/gopass/secrets/akv.go

```diff
@@ -0,0 +1,319 @@
+package secrets
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/gopasspw/gopass/internal/set"
+	"golang.org/x/exp/maps"
+)
+
+var kvSep = ": "
+
+// AKV is the new Key-Value implementation that will replace KV.
+type AKV struct {
+	password string
+	kvp      map[string][]string
+
+	raw strings.Builder
+
+	fromMime bool
+}
+
+// NewAKV creates a new AKV instances.
+func NewAKV() *AKV {
+	a := &AKV{
+		kvp: make(map[string][]string),
+		raw: strings.Builder{},
+	}
+	a.raw.WriteString("\n")
+
+	return a
+}
+
+// NewKVWithData returns a new KV secret populated with data.
+func NewAKVWithData(pw string, kvps map[string][]string, body string, converted bool) *AKV {
+	kv := NewAKV()
+	kv.password = pw
+	kv.fromMime = converted
+
+	kv.raw.WriteString(pw)
+	kv.raw.WriteString("\n")
+
+	for k, vs := range kvps {
+		for _, v := range vs {
+			_ = kv.Add(k, v)
+		}
+	}
+
+	kv.raw.WriteString(body)
+
+	return kv
+}
+
+// Bytes returns the raw string as bytes.
+func (a *AKV) Bytes() []byte {
+	return []byte(a.raw.String())
+}
+
+// Keys returns all the parsed keys.
+func (a *AKV) Keys() []string {
+	return set.Sorted(maps.Keys(a.kvp))
+}
+
+// Get returns the value of the requested key, if found.
+func (a *AKV) Get(key string) (string, bool) {
+	key = strings.ToLower(key)
+
+	if v, found := a.kvp[key]; found {
+		return v[0], true
+	}
+
+	return "", false
+}
+
+// Values returns all values for that key.
+func (a *AKV) Values(key string) ([]string, bool) {
+	key = strings.ToLower(key)
+
+	v, found := a.kvp[key]
+
+	return v, found
+}
+
+// Set writes a single key.
+func (a *AKV) Set(key string, value any) error {
+	// if it's new key we can just append it at the end
+	if _, found := a.kvp[key]; !found {
+		return a.Add(key, value)
+	}
+
```

Comment:
This one isn't specifying `key = strings.ToLower(key)` unlike all the other ones